enhancements/subscription-content/subscription-injection.md

```diff
@@ -0,0 +1,397 @@
+- Gather feedback for namespace-restricted subscription bundles
+- Release as a tech preview, official OLM operator
+
+##### Tech Preview -> GA
+
+- All APIs reach `v1` stability
+- Projected resource CSI driver reaches GA
+- Security audit for these components and the projected resource CSI driver
+- Scale testing (with potential mitigations like `HorizontalPodAutoscalers`)
+- Disruption testing (with potential mitigations like `PodDisruptionBudgets`)
+
+### Upgrade / Downgrade Strategy
+
+Upgrades and downgrades will be handled via OLM. Each released version will specify:
+
+- Minimum supported version of the projected resource CSI driver
+- Minimum supported kube version of the cluster.
+
+### Version Skew Strategy
+
+How will the component handle version skew with other components? What are the guarantees? Make sure
+this is in the test plan.
+
+Consider the following in developing a version skew strategy for this enhancement:
+
+- During an upgrade, we will always have skew among components, how will this impact your work?
+- Does this enhancement involve coordinating behavior in the control plane and in the kubelet? How
+  does an n-2 kubelet without this feature available behave when this feature is used?
+- Will any other components on the node change? For example, changes to CSI, CRI or CNI may require
+  updating that component before the kubelet.
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation History`.
+
+## Drawbacks
+
+- Since this is being deliverd via OLM, customers will need to opt into capabilities that they may
+  expect by default.
+- Users will have to specify on a per-workload basis which subscriptions should be added, increasing
+  toil.
+- Cluster admins need to manually obtain the entitlement keys and `redhat.repo` configurations.
+
+## Alternatives
+
+### Document what we have
+
+Today, any UBI/RHEL-based container can access subscription content if the `redhat.repo` file is
+mounted into `/run/secrets`, and the entitlement keys are mounted into
+`/run/secrets/etc-pki-entitlement`. This is not well documented in OpenShift. Customers will need to
+obtain entitlement keys and `redhat.repo` configurations on their own, and update pods on their own.
+This would then be repeated in every desired namespace.
+
+Builds have their own docs since they use a separate mechanism to add Secrets and ConfigMaps into
+the build context.
+
+### Mount subscription bundles through the node
+
+As a work-around, customers can create a MachineConfig that adds the same entitlement keys to
+/etc/pki/entitlement/ on every node [1]. CRI-O and buildah are configured by default to mount these
+entitlement keys into all running containers.
+
+There are two main downsides to this approach:
+
+1. Nodes need to be restarted when the entitlement keys are added or updated.
+2. Only one `redhat.repo` definitions is available, and should be tied to the version of RHEL the
```

Comment:
What uses this `redhat.repo` file?  As far as I can tell, RHEL 6, 7, and 8 containers don't need or use it.

Reply:
I believe Satellite nodes/clusters need to override `redhat.repo`. 

I will update this EP so we oterhwise use what is provided within the rhel-based containers.